doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,258 @@
+# Remove fine-grained features
+# ============================
+# We use morphological filters to sharpen the mask and focus on the dirt
+# spots. The two fundamental morphological operators are *dilation* and
+# *erosion*, where dilation (resp. erosion) sets the pixel to the brightest
+# (resp. darkest) value of the neighborhood defined by a structuring element
+# (footprint).
+#
+# Here, we use the ``diamond`` function from the ``morphology`` module to
+# create a diamond-shaped footprint.
+# An erosion followed by a dilation is called an *opening*.
+# First, we apply an opening filter, in order to remove small objects and thin
+# lines, while preserving the shape and size of larger objects.
+
+footprint = ski.morphology.diamond(3)
+mask_open = ski.morphology.opening(mask_2, footprint)
+plot_comparison(mask_2, mask_open, "mask before", "after opening")
+
+#####################################################################
+# Since "opening" an image starts with an erosion operation, bright regions
+# which are smaller than the structuring element have been removed.
+# When applying an opening filter, tweaking the footprint parameter lets us
+# control how fine-grained the removed features are. For example, if we used
+# ``footprint = ski.morphology.diamond(1)`` in the above, we could see that
+# only smaller features would be filtered out, hence retaining more spots in
+# the mask.
+# Next, we can make the detected areas wider by applying a dilation filter:
+
+mask_dilate = ski.morphology.dilation(mask_open, footprint)
+plot_comparison(mask_open, mask_dilate, "before", "after dilation")
+
+#####################################################################
+# Dilation enlarges bright regions and shrinks dark regions.
+# Notice how, indeed, the white spots have thickened.
+
+#####################################################################
+# Inpaint each frame separately
+# =============================
+# We are now ready to apply inpainting to each frame. For this we use function
+# ``inpaint_biharmonic`` from the ``restoration`` module. It implements an
+# algorithm based on biharmonic equations.
+# This function takes two arrays as inputs:
+# The image to restore and a mask (with same shape) corresponding to the
+# regions we want to inpaint.
+
+image_seq_inpainted = np.zeros(image_seq.shape)
+
+for i in range(image_seq.shape[0]):
+    image_seq_inpainted[i] = ski.restoration.inpaint_biharmonic(
+        image_seq[i],
+        mask_dilate
+    )
+
+#####################################################################
+# Let us visualize one restored image, where the dirt spots have been
+# inpainted. First, we find the contours of the dirt spots (well, of the mask)
+# so we can draw them on top of the restored image:
+
+contours = ski.measure.find_contours(mask_dilate)
+
+# Gather all (row, column) coordinates of the contours
+x = []
+y = []
+for contour in contours:
+    x.append(contour[:, 0])
+    y.append(contour[:, 1])
```

Comment:
```suggestion
x, y = zip(*((c[:, 0], c[:, 1]) for c in contours))
```
Probably less readable?

Reply:
Yes, but good to teach perhaps? What I've done in the past was to include such compact expressions in the narrative description just 'fyi' :nerd_face:

Reply:
Done as a comment within the code block: 212dd3ed21df4